test/12-device-config.spec.ts

```diff
@@ -27,190 +28,302 @@ const splashImageBackend = new SplashImage();
+	const deviceTypeJson = path.join(bootMountPoint, 'device-type.json');
+	const osRelease = path.join(constants.rootMountPoint, '/etc/os-release');
+
 	let logSpy: SinonSpy;
 
 	before(async () => {
+		// disable log output during testing
+		stub(log, 'debug');
+		stub(log, 'warn');
+		stub(log, 'info');
+		stub(log, 'event');
+		stub(log, 'success');
 		logSpy = spy(logger, 'logSystemMessage');
 		await prepare();
+
+		// clear memoized data from config
+		fnSchema.deviceType.clear();
+		fnSchema.deviceArch.clear();
 	});
 
 	after(() => {
-		logSpy.restore();
+		restore();
 	});
 
 	afterEach(() => {
+		// Restore stubs
 		logSpy.resetHistory();
 	});
 
-	it('correctly parses a config.txt file', async () => {
-		// Will try to parse /test/data/mnt/boot/config.txt
-		await expect(
-			// @ts-ignore accessing private value
-			deviceConfig.getBootConfig(configTxtBackend),
-		).to.eventually.deep.equal({
-			HOST_CONFIG_dtparam: '"i2c_arm=on","spi=on","audio=on"',
-			HOST_CONFIG_enable_uart: '1',
-			HOST_CONFIG_disable_splash: '1',
-			HOST_CONFIG_avoid_warnings: '1',
-			HOST_CONFIG_gpu_mem: '16',
+	describe('formatConfigKeys', () => {
+		it('accepts RESIN_ and BALENA_ variables', async () => {
+			return expect(
+				deviceConfig.formatConfigKeys({
+					FOO: 'bar', // should be removed
+					BAR: 'baz', // should be removed
+					RESIN_SUPERVISOR_LOCAL_MODE: 'false', // any device
+					BALENA_SUPERVISOR_OVERRIDE_LOCK: 'false', // any device
+					BALENA_SUPERVISOR_POLL_INTERVAL: '100', // any device
+					RESIN_HOST_CONFIG_dtparam: 'i2c_arm=on","spi=on","audio=on', // config.txt backend
+					RESIN_HOST_CONFIGFS_ssdt: 'spidev1.0', // configfs backend
+					BALENA_HOST_EXTLINUX_isolcpus: '1,2,3', // extlinux backend
+				}),
+			).to.deep.equal({
+				SUPERVISOR_LOCAL_MODE: 'false',
+				SUPERVISOR_OVERRIDE_LOCK: 'false',
+				SUPERVISOR_POLL_INTERVAL: '100',
+				HOST_CONFIG_dtparam: 'i2c_arm=on","spi=on","audio=on',
+				HOST_CONFIGFS_ssdt: 'spidev1.0',
+				HOST_EXTLINUX_isolcpus: '1,2,3',
+			});
+		});
+	});
+
+	describe('getDefaults', () => {
+		it('returns default configuration values', () => {
+			const conf = deviceConfig.getDefaults();
+			return expect(conf).to.deep.equal({
+				HOST_FIREWALL_MODE: 'off',
+				HOST_DISCOVERABILITY: 'true',
+				SUPERVISOR_VPN_CONTROL: 'true',
+				SUPERVISOR_POLL_INTERVAL: '900000',
+				SUPERVISOR_LOCAL_MODE: 'false',
+				SUPERVISOR_CONNECTIVITY_CHECK: 'true',
+				SUPERVISOR_LOG_CONTROL: 'true',
+				SUPERVISOR_DELTA: 'false',
+				SUPERVISOR_DELTA_REQUEST_TIMEOUT: '59000',
+				SUPERVISOR_DELTA_APPLY_TIMEOUT: '0',
+				SUPERVISOR_DELTA_RETRY_COUNT: '30',
+				SUPERVISOR_DELTA_RETRY_INTERVAL: '10000',
+				SUPERVISOR_DELTA_VERSION: '2',
+				SUPERVISOR_INSTANT_UPDATE_TRIGGER: 'true',
+				SUPERVISOR_OVERRIDE_LOCK: 'false',
+				SUPERVISOR_PERSISTENT_LOGGING: 'false',
+				SUPERVISOR_HARDWARE_METRICS: 'true',
+			});
+		});
+	});
+
+	describe('config.txt', () => {
```

Comment:
One more thing I noticed, when I ran test locally, following all tests passing, test/data/mnt/boot/config.txt gets changed to the following:
```
dtoverlay=balena-fin
```
Whereas before it was a 1195-line file. I would guess something goes wrong with the setup/teardown here, maybe? Alternatively, something going on with the balena fin test suite starting on line 411 of this spec.

Reply:
hmmm, I might have missed something in the setup/teardown, let me check

Reply:
I still get the issue if I skip this test suite, still trying to figure out where the issue is

Reply:
I could not figure out where the config.txt was getting removed, it is some weird interaction between the target state tests and the API tests. It doesn't seem to affect subsequent tests so I'm leaving it for now.